### PR TITLE
Chore/4090 quality gates

### PR DESCRIPTION
**Pull request by bruceharris**
This PR

- adds type checking, linting, and format checking to the CI build, via the new `npm run verify` command which runs all of the above as well as tests
- updates `tsconfig` to ignore library typing errors, as we currently have a number of errors in the OpenFin type declaration files
- fixes all typing errors/warnings, and resolves almost all explicit usages of `any`
- revamps the eslint configuration, as the previous configuration was not doing much
  - note, there are some plugins that were specified in package.json that were not actually used; some of those plugins would be useful, but in the current PR we are going with a more basic configuration given the plethora of linting errors that needed to be resolved without those additional rules
- fixes all linting errors/warnings
- formats all files to align with the format check that is now in the CI build
- removes `React.FC` as in recent years the React community has largely reached consensus
that favors avoiding the use of `React.FC`, and the "recommended" eslint config requires a different form of specifying props regardless
  - see https://github.com/gndelia/codemod-replace-react-fc-typescript and https://github.com/facebook/create-react-app/pull/8177 for discussion

Some linting rules, including those that enforce strict typing, have been disabled for specific files/lines for the following scenarios, as these are either non-production code, or require more significant effort to remediate:
- tests
- mock data
- storybook files
- styleguide files
- files using platform APIs such as OpenFin or Finsemble, where typing support is somewhat lacking
- where there were existing TODOs to resolve tech debt

### CODE REVIEW

File: src/client/src/App/LiveRates/Tile/Tile.rfq.test.tsx [hunk 1]
```diff
@@ -216,7 +218,7 @@ describe("Tile/rfq", () => {
       response$.complete()
     })
 
-    expect(screen.queryByRole("alert")!.textContent).toEqual(
+    expect(screen.getByRole("alert").textContent).toEqual(
```
Comment (algreasley) [inline src/client/src/App/LiveRates/Tile/Tile.rfq.test.tsx:221]: gonna crash the test, obvs, if alert not there ... but may be no doubt, so ok with get?

File: src/client/src/App/Trades/TradesGrid/TradesGrid.tsx [hunk 2]
```diff
@@ -137,7 +137,7 @@ export const TradesGridInner = <Row extends Trade>({
                           ? "right"
                           : "left")
                       }
-                      crossed={isRowCrossed?.(row as Row)}
+                      crossed={isRowCrossed?.(row as unknown as Row)}
```
Comment (algreasley) [inline src/client/src/App/Trades/TradesGrid/TradesGrid.tsx:140]: oof 😄 always makes me wince (have had to do it a few times myself ..)

File: src/client/src/OpenFin/Snapshots/Snapshots.tsx [hunk 3]
```diff
@@ -38,11 +37,11 @@ const useSnapshots = () => {
 }
 
 interface ListProps {
-  snapshots: Record<string, Snapshot>
+  snapshots: Record<string, OpenFin.Snapshot>
```
Comment (algreasley) [inline src/client/src/OpenFin/Snapshots/Snapshots.tsx:40]: may have missed this, but where does the global OpenFin come from .. is that related to package version updates?

File: src/client/src/OpenFin/Window/WindowViewport.tsx [hunk 4]
```diff
@@ -1,3 +1,8 @@
+// disabling these rules for the time being as there are
```
Comment (algreasley) [inline src/client/src/OpenFin/Window/WindowViewport.tsx:1]: as earlier, previous goes at this have led me towards centralising exceptions in overrides in the eslint config, then the various flavours of errors/warnings can be managed and corrected in one commit/PR ..

File: src/client/src/OpenFin/apps/Launcher/utils/openfin.ts [hunk 5]
```diff
@@ -1,19 +1,20 @@
 // TODO - Revisit if/when launcher is cross platform
```
Comment (algreasley) [inline src/client/src/OpenFin/apps/Launcher/utils/openfin.ts:1]: I think this TODO has become defunct 😄

File: src/client/src/OpenFin/apps/Launcher/utils/openfin.ts [hunk 6]
```diff
@@ -1,19 +1,20 @@
...
-import { Application } from "openfin/_v2/main"
 import { Bounds } from "openfin/_v2/shapes/shapes"
```
Comment (algreasley) [inline src/client/src/OpenFin/apps/Launcher/utils/openfin.ts:2]: does this one change now as well?

File: src/client/src/services/limitChecker/limitChecker.openfin.ts [hunk 7]
```diff
@@ -1,3 +1,9 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+// This file uses legacy OpenFin API which is not compatible with the
+// types of the version of the OpenFin lib we are using. This should
+// be fixed, but in the meantime, we are ignoring type checking on
```
Comment (algreasley) [inline src/client/src/services/limitChecker/limitChecker.openfin.ts:4]: raise ticket (and reference here if poss) please :)

File: src/client/src/services/trades/__mocks__/creditTrades.ts [hunk 8]
```diff
@@ -10,9 +10,9 @@ export const mockCreditTrades: CreditTrade[] = [
     counterParty: "J.P. Morgan",
     cusip: "10077LBC9",
     security: "ORCL 4.755 08/15/2026",
-    quantity: "10000",
+    quantity: 10000,
```
Comment (algreasley) [inline src/client/src/services/trades/__mocks__/creditTrades.ts:13]: this is ringing some bells about TODOs in the non-mock code about Hydra data ... ?

File: src/client/src/styleguide/index.tsx [hunk 9]
```diff
@@ -45,6 +45,8 @@ const Styleguide: FC = () => {
       </Helmet>
       <Root>
         <FloatingTools />
+        {/* eslint-disable-next-line @typescript-eslint/ban-ts-comment */}
```
Comment (algreasley) [inline src/client/src/styleguide/index.tsx:48]: ack, this is unfortunate .. anything we can do in future to avoid these messy commented exclusions?

File: src/client/src/styleguide/sections/CoreBranding.tsx [hunk 10]
```diff
@@ -194,8 +196,8 @@ export const Swatch: React.FC<SwatchProps> = ({
       {value}
     </Text>
     {/* <Text fontSize={0.75} textTransform="uppercase" opacity={0.75}>
-      {code}
-    </Text> */}
+    {code}
```
Comment (algreasley) [inline src/client/src/styleguide/sections/CoreBranding.tsx:199]: can we remove this?

File: src/client/tsconfig.json [hunk 11]
```diff
@@ -8,7 +8,7 @@
     "lib": ["DOM", "DOM.Iterable", "ESNext"],
     "types": ["vite/client", "node", "openfin", "jest"],
     "allowJs": false,
-    "skipLibCheck": false,
+    "skipLibCheck": true,
```
Comment (algreasley) [inline src/client/tsconfig.json:11]: yeah, I had to do this testing on command line .. guess we are at the mercy of old OpenFin libs, unless we can specifically exclude that case ?

Comment (bruceharris) [reply src/client/tsconfig.json:11]: I think we can resolve this by upgrading our OF code as a separate tech debt remediation effort

File: src/client/src/styleguide/index.tsx (hunk 9 above)
Comment (bruceharris) [reply src/client/src/styleguide/index.tsx:48]: yes, we can fix the underlying typing issues, but there are a number of them in the styleguide so the effort would be non-trivial, and given the styleguide is not part of the actual application, I made the decision to not address all of them as part of this effort.

File: src/client/src/App/Analytics/PnL/PNLBar.test.tsx [hunk 12]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
```
[the remark this answers is not in the document]
Comment (bruceharris) [reply src/client/src/App/Analytics/PnL/PNLBar.test.tsx:1]: yes, please see my initial comment on the PR where I outlined the general rules I approached this effort with

File: src/client/package.json [hunk 13]
```diff
@@ -38,14 +44,14 @@
...
-    "format": "prettier --write README.md \"src/**/*.{js,jsx,ts,tsx,json,md}\"",
     "generateCod": "hydra-web-codegen -i trading-gateway.hyer -o src/generated",
     "storybook": "start-storybook -p 6006 --no-manager-cache",
     "build-storybook": "build-storybook -o ./dist/storybook"
   },
   "dependencies": {
     "@adaptive/hydra-platform": "^3.120.0",
     "@finos/fdc3": "^1.2.0",
+    "@openfin/core": "^27.71.21",
     "@openfin/workspace": "^6.1.6",
```
[the remark this answers is not in the document]
Comment (bruceharris) [reply src/client/package.json:55]: yes, I was trying not to tackle additional tech debt like that in this PR given the massive scope

File: .github/workflows/tag.yml [hunk 14]
```diff
@@ -49,6 +49,10 @@ jobs:
       - name: Install
         run: npm ci
 
+      - name: Verify Quality (type checking, linting, format checking, tests)
```
[the remark this answers is not in the document]
Comment (bruceharris) [reply .github/workflows/tag.yml:52]: not sure if we allow merge commits, but if we do, a successful run on the branch doesn't guarantee success on master after merge

File: src/client/src/App/Analytics/PnL/PNLBar.test.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (algreasley) [reply src/client/src/App/Analytics/PnL/PNLBar.test.tsx:1]: yep, saw that .. perhaps we can evolve this with some centralised exclusions per another comment .. would appreciate a micro-discussion on this before you head off :)

File: src/client/src/services/limitChecker/limitChecker.openfin.ts (hunk 7 above)
Comment (algreasley) [reply src/client/src/services/limitChecker/limitChecker.openfin.ts:4]: agreed not to reference here given public nature of codebase

**Review by algreasley — APPROVED** [review]

File: src/client/src/App/LiveRates/Tile/Tile.rfq.test.tsx (hunk 1 above)
Comment (bruceharris) [reply src/client/src/App/LiveRates/Tile/Tile.rfq.test.tsx:221]: yes, queryBy* is for when you're expecting no matches https://testing-library.com/docs/queries/about/#types-of-queries

File: src/client/src/OpenFin/Snapshots/Snapshots.tsx (hunk 3 above)
Comment (bruceharris) [reply src/client/src/OpenFin/Snapshots/Snapshots.tsx:40]: it's a namespace declared in `openfin-adapter`
